src/vm/arm64/asmhelpers.S

```diff
@@ -429,8 +429,8 @@ LOCAL_LABEL(CheckCardTable):
     cmp  x15,  x12
 
     ldr  x12, LOCAL_LABEL(wbs_ephemeral_high)
-    ccmp x15, x12, 0x0, ge
-    bgt  LOCAL_LABEL(Exit)
+    ccmp x15, x12, 0x0, hs
```

Comment:
I do not think there was real problem on Unix. We would take the SkipEphemeralCheck path above when the ephemeral checks are disabled.

It does not hurt to fix it though.

Reply:
If you are referring to the `cbz  x12, LOCAL_LABEL(SkipEphemeralCheck)` above, then it doesn't filter out the case we have, since the ephemeral low limit is set to 1 in server GC.

Reply:
If I am reading this correctly, the local copy of the limits is 0. It gets transformed here:

https://github.com/janvorli/coreclr/blob/e3877b29a43fde5600b3e053d7ef472af4cf5123/src/vm/arm64/asmhelpers.S#L261

Reply:
Except `EphemeralCheckEnabled` should be false for gcserver mode
So that code should not run